enhancements/console/dynamic-plugins.md

```diff
@@ -534,6 +602,55 @@ Both `v1` and `v1alpha1` version are supported. `v1alpha1` plugins will get
 converted by the conversion webhook server into `v1` representation.
 Conversion webhook server is part of the `console-operator` pod.
 
+#### Content Security Policy
+For Content Security Policy (CSP) feature to be considered stable, the following
+requests need to be met:
+
+##### 1. Documentation updates
+
+* Update the OpenShift official documentation to include detailed guidelines
+on configuring `ConsolePluginCSP` in the `ConsoleDynamicPlugin` CRD, along with
+recommendations.
+PR link: TBD
+
+##### 2. Release notes updates
+
+* Add CSP feature to release notes.
+  * Pull request link: TBD
+
+##### 3. Extending integration and unit test suite
+
+* Extend integration and unit test suite for console-operator repository:
+    * Standart use case, with a single plugin setting valid sources and
+    validating the final CSP result set in the `console-config.yaml` file.
+    * Edge use cases, such as many plugins defining CSPs with overlapping sources,
+    and validates the final merged CSP result, set in the `console-config.yaml` file.
+    * Pull request link: https://github.com/openshift/console-operator/pull/938
+
+##### 4. Updates to demo plugin
+
+* Update one console-demo-plugin to use the new `ConsolePluginCSP` field
+in order to demonstrate the usage.
+  * Pull request link: TBD
+
+##### 5. Integration tests updates
+
+* Extend integration test suite in console repository CI to check for CSP violations.
+* PR link: [TBD](https://issues.redhat.com/browse/CONSOLE-4279)
+
+##### 6. CI updates to all dynamic plugins
+
+* All the dynamic plugins which are enabled on the cluster by default will
+update their CI to check for CSP violations.
+* PR link: TBD
+
+Currently Console uses `Content-Security-Policy-Report-Only` instead of
+`Content-Security-Policy` header. Due to that the browser will only warn about
+Console CSP violations.
+
+Once all the above listed requests are met, Console will switch will be switched
+to use `Content-Security-Policy` header and start enforcing the policies.
```

Comment:
We also need to give teams outside of Red Hat authoring plugins a chance to update their plugins before we begin enforcing, so minimally that at least one release where we report only. I think it would be better to have a 2-3 release grace period, though.

(But of course we need the API available before then so plugins can adopt it and test.)

Reply:
@spadgett definitely it will need to be at least 2-3 releases. I've rather avoid mentioning that in the doc, since its a bit unclear, instead I've only listed necessary tasks which need to be done before we actually start think about flipping the switch

Reply:
Maybe we say we will stay in report-only mode for at least one release (which doesn't stop us from waiting longer).

Reply:
will add 👍